clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -320,12 +320,27 @@ static bool checkAttrMutualExclusion(Sema &S, Decl *D, const Attr &AL) {
   return false;
 }
 
-static bool checkDeprecatedSYCLAttributeSpelling(Sema &S,
-                                                 const ParsedAttr &Attr) {
-  if (Attr.getScopeName()->isStr("intelfpga"))
-    return S.Diag(Attr.getLoc(), diag::warn_attribute_spelling_deprecated)
-           << "'" + Attr.getNormalizedFullName() + "'";
-  return false;
+void Sema::CheckDeprecatedSYCLAttributeSpelling(const ParsedAttr &A,
```

Comment:
A question for you.   One of the new developers had mentioned that they found it difficult to figure out what the single letter variable names meant in a function; and I think we had changed to using slightly longer spellings to make that clearer.  What is the clang convention, if there is any, for this?  Are single letter variable names considered acceptable in "obvious" situations?

Reply:
Hi @AaronBallman , FYI, when I started contributing to this repo, @pvchupin pointed the [Contibution-guide](https://github.com/intel/llvm/blob/sycl/CONTRIBUTING.md), which suggests a few pointers to be followed while preparing our patch ( See section "Prepare your patch"). One of them lists following the LLVM coding standards and for this specific case [Naming-conventions](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) lists how we could better name variables, functions etc.

```
class VehicleMaker {
  ...
Factory<Tire> F;               // Avoid: a non-descriptive abbreviation.
  Factory<Tire> Factory;      // Better: more descriptive.
  Factory<Tire> TireFactory;  // Even better: if VehicleMaker has more than one kind of factories.
};
```

I am not aware of cases where these guidelines are relaxed by the community, but thought I would just share.

Reply:
Thank you for the links @srividya-sundaram! 

> I am not aware of cases where these guidelines are relaxed by the community, but thought I would just share.

The relaxation I was talking about is in the Introduction of the coding standard document (https://llvm.org/docs/CodingStandards.html#introduction): "If you are extending, enhancing, or bug fixing already implemented code, use the style that is already being used so that the source is uniform and easy to follow."

That said, I agree that more descriptive names would be better in cases where the naming matters for understanding the code.

Reply:
> If you are extending, enhancing, or bug fixing already implemented code, use the style that is already being used so that the source is uniform and easy to follow.

Thanks for pointing this out. I am myself in the midst of updating FE tests to be more readable/maintainable and found updating the non-descriptive variable names exhausting. I could apply this rule for some cases I guess.

Reply:
FWIW, my usual rule of thumb is to use the ridiculously short names for variables and parameters that are only used in a limited scope, and go with more verbose names for member variables or within larger function body scopes.

People tend not to complain about code like:
```
if (const auto *FD = dyn_cast<FunctionDecl>(D)) {
  FD->whatever();
}
```
or similarly restricted contexts. However, I also started back when the project used these short names more exclusively and so I picked up some bad habits. ;-)